src/services/client.ts

```diff
@@ -569,15 +569,21 @@ export default class ClClient {
     }
 
     //AT.CREATE_TEACH_SESSION_FROMHISTORYASYNC
-    async teachSessionFromHistory(appId: string, trainDialog: CLM.TrainDialog, userInput: CLM.UserInput | null, userName: string, userId: string): Promise<CLM.TeachWithHistory> {
+    // filteredDialog = dialog to ignore when checking for conflicting labels
+    async teachSessionFromHistory(appId: string, trainDialog: CLM.TrainDialog, userInput: CLM.UserInput | null, userName: string, userId: string, filteredDialog: string | null): Promise<CLM.TeachWithHistory> {
+        let url = `${this.baseUrl}/app/${appId}/teachwithhistory?username=${userName}&userid=${userId}`
+        if (filteredDialog) {
```

Comment:
This is misleading, it's a dialog id and not a dialog. Would prefer: `filteredDialogId`

It's a bigger change and probably not for this PR but would prefer the name to specify how it's filtering. It doesn't indicate if the filters is only id's given or excluding the id's given. Would make more sense to name is `excludeDialogIds`  so you don't have to remmeber which API's are positive or negative conditions

Reply:
API calls it "filterDialog" so I went with that but I agree I prefer if it says what it is